tokio/src/sync/broadcast.rs

```diff
@@ -310,7 +311,7 @@ struct Shared<T> {
     mask: usize,
 
     /// Tail of the queue. Includes the rx wait list.
-    tail: Mutex<Tail>,
+    tail: RwLock<Tail>,
```

Comment:
`RwLock` implementations tend to be heavier than `Mutex`. In this case, it looks like all reads are of numbers. Another option is to make these cells `AtomicUsize` (or `AtomicU64`) and require writers to these cells to hold the `tail` mutex. Reads can do the atomic read directly.

Reply:
Are you saying that we could make `Tail` fields atomic? That would spare us the need to take a lock in some situations, but the main contention source is adding waiters to the list, which would still have to be done with a lock.